v2/sender.go

```diff
@@ -135,7 +139,9 @@ func (d *Sender) SendMessageBatch(ctx context.Context, messages []*azservicebus.
 	if ctx.Err() != nil {
 		return fmt.Errorf("failed to send message: %w", ctx.Err())
 	}
-	
+
+	d.mu.RLock()
```

Comment:
Just to be clear, this means that calling FailOver has to wait for messages to be sent before failing over.
This make it quite critical to have a timeout on the Send calls, or you can be deadlocked. (you cannot send, and you cannot failover).
Do I understand the code correctly?

Reply:
Yes. Even without the failover, we should always have a timeout to the Sends because they could hang indefinitely.